airflow/gcp/hooks/campaign_manager.py

```diff
@@ -0,0 +1,229 @@
+        :type report_id: str
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .patch(profileId=profile_id, reportId=report_id)
+            .execute(num_retries=self.num_retries)
+        )
+        return response
+
+    def run_report(
+        self, profile_id: str, report_id: str, synchronous: Optional[bool] = None
+    ) -> Any:
+        """
+        Runs a report.
+
+        :param profile_id: The DFA profile ID.
+        :type profile_id: str
+        :param report_id: The ID of the report.
+        :type report_id: str
+        :param synchronous: If set and true, tries to run the report synchronously.
+        :type synchronous: Optional[bool]
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .run(profileId=profile_id, reportId=report_id, synchronous=synchronous)
+            .execute(num_retries=self.num_retries)
+        )
+        return response
+
+    def update(self, profile_id: str, report_id: str) -> Any:
+        """
+        Updates a report.
+
+        :param profile_id: The DFA user profile ID.
+        :type profile_id: str
+        :param report_id: The ID of the report.
+        :type report_id: str
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .update(profileId=profile_id, reportId=report_id)
+            .execute(num_retries=self.num_retries)
+        )
+        return response
+
+    def get(self, file_id: str, profile_id: str, report_id: str) -> Any:
+        """
+        Retrieves a report file.
+
+        :param profile_id: The DFA user profile ID.
+        :type profile_id: str
+        :param report_id: The ID of the report.
+        :type report_id: str
+        :param file_id: The ID of the report file.
+        :type file_id: str
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .files()
+            .get(fileId=file_id, profileId=profile_id, reportId=report_id)
+            .execute(num_retries=self.num_retries)
+        )
+        return response
+
+    def get_report_file(
+        self, file_id: str, profile_id: str, report_id: str
+    ) -> http.HttpRequest:
+        """
+        Retrieves a media part of report file.
+
+        :param profile_id: The DFA user profile ID.
+        :type profile_id: str
+        :param report_id: The ID of the report.
+        :type report_id: str
+        :param file_id: The ID of the report file.
+        :type file_id: str
+        :return: googleapiclient.http.HttpRequest
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .files()
+            .get_media(fileId=file_id, profileId=profile_id, reportId=report_id)
+        )
+        return response
```

Comment:
Can you add the wait for operation method? This is not needed, but it is useful from the point of view of the whole hook.